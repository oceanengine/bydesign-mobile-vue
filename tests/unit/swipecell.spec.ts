import { mount, triggerDrag } from '../index';
import { SwipeCell as NativeSwipeCell } from '@src';
const SwipeCell = NativeSwipeCell as any;
describe('SwipeCell', () => {
  const defaultProps = {
    propsData: {
      leftWidth: 100,
      rightWidth: 100,
    },
    scopedSlots: {
      left: () => 'Left',
      right: () => 'Right',
    },
  };

  const THRESHOLD = 0.15;
  test('drag and show left part', () => {
    const wrapper = mount(SwipeCell, defaultProps);

    triggerDrag(wrapper, 10, 0);
    expect(wrapper).toMatchSnapshot();

    triggerDrag(wrapper, 50, 0);
    expect(wrapper).toMatchSnapshot();

    triggerDrag(wrapper, 500, 0);
    expect(wrapper).toMatchSnapshot();

    triggerDrag(wrapper, 0, 100);
    expect(wrapper).toMatchSnapshot();
  });

  test('drag and show right part', () => {
    const wrapper = mount(SwipeCell, defaultProps);

    triggerDrag(wrapper, -50, 0);
    expect(wrapper).toMatchSnapshot();
  });
  test('on-close prop', async () => {
    let position;
    let instance;

    const wrapper = mount(SwipeCell, {
      ...defaultProps,
      propsData: {
        ...defaultProps.propsData,
        async: true,
      },
      listeners: {
        close(pos, ins) {
          position = pos;
          instance = ins;
        },
      },
    }) as any;

    wrapper.trigger('click');
    expect(position).toEqual(undefined);
    await wrapper.vm.open('left');
    wrapper.find('.byted-swipe-cell__left').trigger('click');
    expect(position).toEqual('left');

    wrapper.find('.byted-swipe-cell__right').trigger('click');
    expect(position).toEqual('right');
    instance.close();
    expect(instance.offset).toEqual(0);
  });

  test('should reset after drag', () => {
    const wrapper = mount(SwipeCell, defaultProps);
    triggerDrag(wrapper, defaultProps['leftWidth'] * THRESHOLD - 1, 0);
    expect(wrapper.vm['offset']).toEqual(0);
  });

  test('disabled prop', () => {
    const wrapper = mount(SwipeCell, {
      propsData: {
        ...defaultProps.propsData,
        disabled: true,
      },
    });

    triggerDrag(wrapper, 50, 0);
    expect(wrapper.vm['offset']).toEqual(0);
  });
  test('render one side', async () => {
    const wrapper = mount(SwipeCell, {
      scopedSlots: {
        left: defaultProps.scopedSlots.left,
      },
    });

    await triggerDrag(wrapper, 100, 0);
    expect(wrapper).toMatchSnapshot();
  });
});
