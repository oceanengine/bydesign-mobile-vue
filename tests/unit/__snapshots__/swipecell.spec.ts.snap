// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwipeCell drag and show left part 1`] = `
<div class="byted-swipe-cell">
  <div class="byted-swipe-cell__wrapper" style="transform: translate3d(0px, 0, 0); transition: .6s cubic-bezier(0.18, 0.89, 0.32, 1);">
    <div class="byted-swipe-cell__left">Left</div>
    <div class="byted-swipe-cell__right">Right</div>
  </div>
</div>
`;

exports[`SwipeCell drag and show left part 2`] = `
<div class="byted-swipe-cell">
  <div class="byted-swipe-cell__wrapper" style="transform: translate3d(0px, 0, 0); transition: .6s cubic-bezier(0.18, 0.89, 0.32, 1);">
    <div class="byted-swipe-cell__left">Left</div>
    <div class="byted-swipe-cell__right">Right</div>
  </div>
</div>
`;

exports[`SwipeCell drag and show left part 3`] = `
<div class="byted-swipe-cell">
  <div class="byted-swipe-cell__wrapper" style="transform: translate3d(0px, 0, 0); transition: .6s cubic-bezier(0.18, 0.89, 0.32, 1);">
    <div class="byted-swipe-cell__left">Left</div>
    <div class="byted-swipe-cell__right">Right</div>
  </div>
</div>
`;

exports[`SwipeCell drag and show left part 4`] = `
<div class="byted-swipe-cell">
  <div class="byted-swipe-cell__wrapper" style="transform: translate3d(0px, 0, 0); transition: .6s cubic-bezier(0.18, 0.89, 0.32, 1);">
    <div class="byted-swipe-cell__left">Left</div>
    <div class="byted-swipe-cell__right">Right</div>
  </div>
</div>
`;

exports[`SwipeCell drag and show right part 1`] = `
<div class="byted-swipe-cell">
  <div class="byted-swipe-cell__wrapper" style="transform: translate3d(0px, 0, 0); transition: .6s cubic-bezier(0.18, 0.89, 0.32, 1);">
    <div class="byted-swipe-cell__left">Left</div>
    <div class="byted-swipe-cell__right">Right</div>
  </div>
</div>
`;

exports[`SwipeCell render one side 1`] = `
<div class="byted-swipe-cell">
  <div class="byted-swipe-cell__wrapper" style="transform: translate3d(0px, 0, 0); transition: .6s cubic-bezier(0.18, 0.89, 0.32, 1);">
    <div class="byted-swipe-cell__left">Left</div>
  </div>
</div>
`;
